File: eval/fixtures/rails-pms/app/models/space.rb
# A bookable space (room, suite, cabin…).
#
# Spaces are archived rather than deleted so past reservations keep pointing at
# something real.
class Space < ApplicationRecord
  # The kinds of bookable space a property can offer, plus their display labels.
  KINDS = %w[room suite cabin villa tent other].freeze
  KIND_LABELS = {
    "room" => "Room",
    "suite" => "Suite",
    "cabin" => "Cabin",
    "villa" => "Villa",
    "tent" => "Tent",
    "other" => "Space",
  }.freeze

  STATUSES = %w[active archived].freeze

  has_many :reservations, dependent: :nullify

  validates :name, presence: true, uniqueness: true, length: { maximum: 80 }
  validates :kind, inclusion: { in: KINDS }
  validates :capacity,
            numericality: {
              only_integer: true,
              greater_than_or_equal_to: 1,
              less_than_or_equal_to: 40,
            }
  validates :status, inclusion: { in: STATUSES }

  scope :only_active, -> { where(status: "active") }

  # Every space, active first then alphabetical.
  def self.list
    order(Arel.sql("status ASC"), Arel.sql("name ASC"))
  end

  def active?
    status == "active"
  end

  def kind_label
    KIND_LABELS.fetch(kind, "Space")
  end
end

File: eval/fixtures/laravel-pms/app/Models/Space.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * A bookable space (room, suite, cabin…).
 *
 * Spaces are archived rather than deleted so past reservations keep pointing at
 * something real.
 *
 * @property int $id
 * @property string $name
 * @property string $kind
 * @property int $capacity
 * @property int|null $rate_cents
 * @property string $status
 * @property string|null $notes
 */
class Space extends Model
{
    /** Only a created_at timestamp — the domain has no updated_at. */
    const UPDATED_AT = null;

    protected $fillable = [
        'name',
        'kind',
        'capacity',
        'rate_cents',
        'status',
        'notes',
    ];

    protected $attributes = [
        'kind' => 'room',
        'capacity' => 2,
        'status' => 'active',
    ];

    protected function casts(): array
    {
        return [
            'capacity' => 'integer',
            'rate_cents' => 'integer',
            'created_at' => 'datetime',
        ];
    }

    /** @return HasMany<Reservation, $this> */
    public function reservations(): HasMany
    {
        return $this->hasMany(Reservation::class);
    }
}
